### PR TITLE
Deprecate LLVM support

### PR DESCRIPTION
**Pull request by sluongng**
This PR includes 2 commits

```
remove llvm toolchain

Since the beginning, we have added llvm 8.0.0 targeting ubuntu 16.04
release as an alternative to gcc.
However, this toolchain has not been popular among our users and
therefore, has not receive a lot of love. The current version is fairly
out of date vs the latest release.

Let's remove this toolchain entirely and guide folks who are interested
in LLVM toolchain toward a better alternative:
https://github.com/bazel-contrib/toolchains_llvm

---

Platform: switch constraint from clang to gcc

Our toolchain and container image uses gcc by default.
So let's make sure our platform reflects the correct constraint value.

In practice, this constraint value is not commonly used. It's mostly
there to help differentiate our remote-only toolchain with a local
Bazel-setup toolchain. This helps avoid us trying to use the remote
toolchain during local build.
```

### CODE REVIEW

File: rules.bzl [hunk 1]
```diff
@@ -84,7 +78,7 @@ def _buildbuddy_toolchain_impl(rctx):
 
     # Repository implementation functions can be restarted, keep expensive ops at the end.
     if (rctx.attr.llvm):
```
Comment (fmeum) [inline rules.bzl:80]:
```suggestion
    if rctx.attr.llvm:
```

File: templates/BUILD.tpl [hunk 2]
```diff
@@ -121,6 +120,16 @@ alias(
 
 ## CC
 
+filegroup(
+    name = "empty",
+    srcs = [],
+)
+
+filegroup(
```
Comment (fmeum) [inline templates/BUILD.tpl:128]: Are you keeping this intentionally?

**Review by fmeum — APPROVED** [review]

File: templates/BUILD.tpl (hunk 2 above)
Comment (sluongng) [reply templates/BUILD.tpl:128]: Yeah it seems like these 2 filegroups are used by other toolchains here, including the ubuntu gcc and windows toolchains. So I moved them to the top of the file as a "common CC" section.

**Review by bduffany — COMMENTED** [review]
This LGTM but will defer to Siggi since he might have more context on this LLVM toolchain.

**Review by siggisim — APPROVED** [review]

**Comment by sluongng** [comment]
tested this in buildbuddy.git with 

```
common --override_repository=io_buildbuddy_buildbuddy_toolchain=/Users/sluongng/work/buildbuddy/buildbuddy-toolchain
```

**Pull request merged**